waspc/data/Generator/templates/sdk/wasp/auth/forms/Auth.tsx

```diff
@@ -55,10 +41,10 @@ function Auth ({ state, appearance, logo, socialLayout = 'horizontal', additiona
   const [successMessage, setSuccessMessage] = useState<string | null>(null);
   const [isLoading, setIsLoading] = useState(false);
 
-  // TODO(matija): this is called on every render, is it a problem?
-  // If we do it in useEffect(), then there is a glitch between the default color and the
-  // user provided one.
-  const customTheme = createTheme(appearance ?? {})
+  const customStyle = useMemo((): CSSProperties => ({
+    ...tokenObjToCSSVars('color', appearance?.colors || {}),
+    ...tokenObjToCSSVars('font-size', appearance?.fontSizes || {}),
+  }), [appearance])
```

Comment:
Nitpick: since we typed `tokenObjToCSSVars` it's okay to drop the type here and I'd use `??`.

```suggestion
  const customStyle = useMemo(() => ({
    ...tokenObjToCSSVars('color', appearance?.colors ?? {}),
    ...tokenObjToCSSVars('font-size', appearance?.fontSizes ?? {}),
  }), [appearance])
```